### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by ssaviak**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,80 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: string,
```
Comment (Xilston) [inline src/figuresClasses.ts:6]: You have created type for `shape (Shape)`  and you use if at 6 line, but you didn't use `Color `for `color `in `Figure `interface.

**Review by Xilston — APPROVED** [review]
Nice Job, just a small note.

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,80 @@
...
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 0 || this.b < 0 || this.c < 0) {
```
Comment (DanilWeda) [inline src/figuresClasses.ts:19]: as alternative*
```suggestion
    if ([this.a, this.b, this.b].every( side => side > 0) {
```

**Review by DanilWeda — APPROVED** [review]
Good job! 
Approved!

**Pull request closed without merging**